### PR TITLE
Fix calls to pods on the same node

### PR DESCRIPTION
**Pull request by howardjohn**

### CODE REVIEW

File: src/proxy/outbound.rs [hunk 1]
```diff
@@ -232,11 +232,14 @@ impl OutboundConnection {
...
-            // so we need to explicitly send it to ourselves.
+            // Sending to a node on the same node (ourselves).
             // In the future this could be optimized to avoid a full network traversal.
             req.request_type = RequestType::DirectLocal;
-            req.gateway = "127.0.0.1:15088".parse().unwrap();
+            // We would want to send to 127.0.0.1:15008 in theory. However, the inbound listener
+            // expects to lookup the desired certificate based on the destination IP. If we send directly,
+            // we would try to lookup an IP for 127.0.0.1.
+            // Instead, we send to the actual IP, but iptables in the pod ensures traffic is redirected to 15008.
+            req.gateway = SocketAddr::from((req.gateway.ip(), 15088));
```
Comment (hzxuzhonghu) [inline src/proxy/outbound.rs:242]: 15008？

Comment (howardjohn) [reply src/proxy/outbound.rs:242]: No, 15088 is setup to redirect back to 15008.

This is not a great hack but expediant way to get things working and it matches what we do in Envoy today (hence why the iptables rule exists today).

Comment (hzxuzhonghu) [reply src/proxy/outbound.rs:242]: got it

Comment (howardjohn) [reply src/proxy/outbound.rs:242]: long term we can probably just do it in memory since we are calling ourselves

Comment (hzxuzhonghu) [reply src/proxy/outbound.rs:242]: how about `req.gateway.set_port`

Comment (howardjohn) [reply src/proxy/outbound.rs:242]: req.gateway is not `mut` so would need more refactoring, but could be done.

I think there is an issue to refactor this code a bit

**Review by hzxuzhonghu — APPROVED** [review]
So lgtm

**Review by stevenctl — APPROVED** [review]

**Pull request merged**